### PR TITLE
combine all the necessary namespace-based resource configure together for easier installation

### PR DESCRIPTION
**Pull request by xfate123**
integrated manifest file. Already test in local for installation process

### CODE REVIEW

**Comment by xfate123** [comment]
I signed it!

**Comment by xfate123** [comment]
@googlebot  I signed it

**Comment by xfate123** [comment]
@googlebot I signed it!

**Comment by xfate123** [comment]
@googlebot I signed it!

File: xgboost-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: xgboost-operator
+  namespace: xgboost-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: mpi-operator
```
Comment (merlintang) [inline xgboost-operator.yaml:170]: ?

File: xgboost-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - get
+- apiGroups:
+  - kubeflow.org
+  resources:
+  - mpijobs
```
Comment (merlintang) [inline xgboost-operator.yaml:142]: ?

File: xgboost-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,201 @@
+#--------------------------------------------------
+#- Single configuration deployment YAML for xgboost-Operator
+#- Include:
+#      CRD deployment
+#      Namespace deployment
+#      RBAC deployment
+#      xgboost-operator container deployment set     
+# --------------------------------------------------
+---
```
Comment (merlintang) [inline xgboost-operator.yaml:9]: this is not namespace based ? I do not see the namespace configure

Comment (xfate123) [reply xgboost-operator.yaml:9]: namespace configure is at line 45

File: xgboost-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                Worker: 
+                  properties:
+                    replicas:
+                      type: integer
+                      minimum: 1
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: xgboost-operator
```
Comment (merlintang) [inline xgboost-operator.yaml:48]: namespace would have a better name rather than operator name. 
for example, test-namespace or my-namespace

Comment (xfate123) [reply xgboost-operator.yaml:48]: sure, I will make some changes

Comment (xfate123) [reply xgboost-operator.yaml]: already changed, appreciate your further review

**Comment by xfate123** [comment]
@terrytangyuan  For sure, I will try a better pr label. Thanks so much for your advice

File: xgboost-operator.yaml (hunk 3 above)
Comment (Jeffwan) [reply xgboost-operator.yaml:13]: This project is generated by kubebuilder and user supposed to use `make` to deploy the manifests. 
Another option is to use `kustomize` to build the manifest and generate one yaml file. 

`my-namepsace` doesn't sounds like a reasonable name. It's better to use `xgboost-operator` or `kubeflow`

Comment (xfate123) [reply xgboost-operator.yaml:13]: Thank you so much for your advice. kustomize sounds like a better option

Comment (merlintang) [reply xgboost-operator.yaml:13]: in the production, we do not have to use the kubebuild, we just need to use the yaml to deploy the controller. Then the namespace would be more meaningful, for example, the namespace might be related to user group information, then xgboost-operator can be the name for the namespace.

**Pull request closed without merging**